src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/RequestBookingService.kt

```diff
@@ -0,0 +1,188 @@
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.Court
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.Notification
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.Prison
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.ReferenceCode
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.Appointment
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AppointmentType
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.BookingType
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.RequestVideoBookingRequest
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.UnknownPrisonerDetails
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.CourtRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.NotificationRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.ReferenceCodeRepository
+
+@Service
+class RequestBookingService(
+  private val emailService: EmailService,
+  private val contactsService: ContactsService,
+  private val appointmentsService: AppointmentsService,
+  private val courtRepository: CourtRepository,
+  private val prisonRepository: PrisonRepository,
+  private val referenceCodeRepository: ReferenceCodeRepository,
+  private val notificationRepository: NotificationRepository,
+  private val locationsInsidePrisonClient: LocationsInsidePrisonClient,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun request(request: RequestVideoBookingRequest, username: String) {
+    when (request.bookingType!!) {
+      BookingType.COURT -> sendCourtRequestEmails(request, username)
+      BookingType.PROBATION -> log.info("TODO - send probation request emails.")
+    }
+  }
+
+  private fun sendCourtRequestEmails(request: RequestVideoBookingRequest, username: String) {
+    appointmentsService.checkCourtAppointments(request.prisoner().appointments, request.prisoner().prisonCode!!)
+
+    val (pre, main, post) = getCourtAppointments(request.prisoner())
+
+    val prison = prisonRepository.findByCode(request.prisoner().prisonCode!!)
+      ?.also { require(it.enabled) { "Prison with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Prison with code ${request.prisoner().prisonCode} not found")
+
+    val court = courtRepository.findByCode(request.courtCode!!)
+      ?.also { require(it.enabled) { "Court with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Court with code ${request.courtCode} not found")
+
+    val hearingType = referenceCodeRepository.findByGroupCodeAndCode("COURT_HEARING_TYPE", request.courtHearingType!!.toString())!!
```

Comment:
Can we throw an exception with a bit of context here instead of !! at the end e.g. `NPE "Court hearing type {type not found}"` ?